File: DS Week 1&2/Algorithms/InsertionSorting/InsertionSorting.java
public class InsertionSorting {
    public static void main(String[] args) {
        int[] arr={30,2,41,56,12,11,22,41};
        int [] sortedArray=insertionSorting(arr);
        for(int i=0;i<sortedArray.length;i++){
            System.out.println(sortedArray[i]);
        }
        
        
    }
    static int [] insertionSorting(int [] array){
        for(int i=0;i<array.length;i++){
            int j=i-1;
            int current=array[i];
            while(j>=0 && current < array[j]){
                array[j+1]=array[j];
                j--;
            }
            array[j+1]=current;
        } 
        return array;
    }
}

File: DS Week 1&2/Code Evolution/Sorting/insertionSorting.js
let arr=[5,1,3,4,45,5,6,3,6,743,35,45]
console.log(insertionSorting(arr));
function insertionSorting(arr){
    for (let i=1;i<arr.length;i++){
        let insertingElement=arr[i]
        for(let j=i-1;j>=0;j--){
            if(insertingElement<arr[j]){
                arr[j+1]=arr[j]
            }
            else{
                arr[j+1]=insertingElement
                break;
            }
            if(j==0){
                arr[j]=insertingElement
            }
        }
    }
    return arr
}